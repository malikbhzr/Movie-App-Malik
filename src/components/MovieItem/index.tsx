import React from 'react';
import { Movie } from '../../model/MoviesModel/Movie';
import {
  TransactionStyle,
  ImageContainer,
  TitleStyle,
} from './styled';

interface MoviItemProps {
  movie: Movie;
  nav: () => void,
}

export default function MoviItem({
    movie,
    nav
}: MoviItemProps) {


  const showItem = () => {
    if(movie){
      return (
        <TransactionStyle testID="cardItem" onPress={() => nav('DetailScreen', { itemId: movie.imdbID})}>
          <ImageContainer  source={{ uri: movie.Poster }} />
          <TitleStyle>{movie.Title}</TitleStyle>
        </TransactionStyle>
      );
    }
  }

  return showItem();
}