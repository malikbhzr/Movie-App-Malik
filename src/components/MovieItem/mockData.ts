export const MockItem = {
    Title: 'Lego Marvel Super Heroes', 
    Year: '2013', imdbID: 'tt2620204', 
    Type: 'game', 
    Poster: 'https://m.media-amazon.com/images/M/MV5BOTA5ODA2NTI2M15BMl5BanBnXkFtZTgwNTcxMzU1MDE@._V1_SX300.jpg',
}