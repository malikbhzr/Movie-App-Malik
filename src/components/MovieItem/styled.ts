import styled from 'styled-components/native';

export const TransactionStyle = styled.TouchableOpacity`
  margin-horizontal: 16px;
  margin-top: 16px;
`;

export const ImageContainer = styled.Image`
  width: 90px;
  height: 150px;
  background-color: #ccc;
  border-radius: 6px;
`;

export const TitleStyle = styled.Text`
  font-size: 16px;
  margin-top: 10px;
  color: #fff;
  margin-left: 5px;
`;