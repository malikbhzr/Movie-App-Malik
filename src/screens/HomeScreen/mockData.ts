export const MockList =
    {
        movies:{
            Search: [{
                Title: "Captain Marvel",
                Year: "2019",
                imdbID: "tt4154664",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
            }, {
                Title: "Captain Marvel",
                Year: "2019",
                imdbID: "tt4154664",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BMTE0YWFmOTMtYTU2ZS00ZTIxLWE3OTEtYTNiYzBkZjViZThiXkEyXkFqcGdeQXVyODMzMzQ4OTI@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: All Hail the King",
                Year: "2014",
                imdbID: "tt3438640",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BZGFkMTZkMDQtNzM4Yy00YWEwLTkzOWEtZTMyNDRlNmJhYWJhXkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: Item 47",
                Year: "2012",
                imdbID: "tt2247732",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BMjNlMzAxNmQtOGEwZi00NTEyLWI0NWYtMTlhNmE2YTA3ZDVhXkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: Agent Carter",
                Year: "2013",
                imdbID: "tt3067038",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BZDIwZTM4M2QtMWFhYy00N2VmLWFlMjItMzI3NjBjYTc0OTMxXkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: A Funny Thing Happened on the Way to Thor's Hammer",
                Year: "2011",
                imdbID: "tt2011109",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BYmVlYTg3N2QtMWM2OS00YWQyLWI2M2MtMDc0ZjBkZjk1MTY3XkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Marvel One-Shot: The Consultant",
                Year: "2011",
                imdbID: "tt2011118",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BNGE4YjU5MDAtYzYzMC00M2RlLTk0NDgtNDU1MjgyMGI0MjI3XkEyXkFqcGdeQXVyNTE1NjY5Mg@@._V1_SX300.jpg"
            }, {
                Title: "Lego Marvel Super Heroes",
                Year: "2013",
                imdbID: "tt2620204",
                Type: "game",
                Poster: "https://m.media-amazon.com/images/M/MV5BOTA5ODA2NTI2M15BMl5BanBnXkFtZTgwNTcxMzU1MDE@._V1_SX300.jpg"
            }, {
                Title: "Adventures of Captain Marvel",
                Year: "1941",
                imdbID: "tt0033317",
                Type: "movie",
                Poster: "https://m.media-amazon.com/images/M/MV5BNjg0NTk3NjUyNF5BMl5BanBnXkFtZTgwNDQ5MjM1MjE@._V1_SX300.jpg"
            }, {
                Title: "Marvel: Ultimate Alliance",
                Year: "2006",
                imdbID: "tt0814207",
                Type: "game",
                Poster: "N/A"
            }],
        },
        totalResults: 151,
        Response: "True"
    }