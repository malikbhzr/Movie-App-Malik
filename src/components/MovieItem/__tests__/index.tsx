import React from 'react';
import MovieItem from '../';
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow, ShallowWrapper } from 'enzyme';
import { MockItem } from '../mockData';

Enzyme.configure({ adapter: new Adapter() });

describe("Movie item", () => {

  describe("render with mock data", () => {
    let item = MockItem;
    const movie={item}
    const navigation = jest.fn();
    let wrapper: ShallowWrapper;
    wrapper = shallow(<MovieItem movie={movie} nav={navigation}  />);
  
    it("should render with mock data", () => {
        expect(wrapper).toMatchSnapshot()
    });
  })
  
});