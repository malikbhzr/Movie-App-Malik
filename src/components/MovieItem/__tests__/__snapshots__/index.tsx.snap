// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie item Movie item should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginHorizontal": 16,
      "marginTop": 16,
      "opacity": 1,
    }
  }
  testID="cardItem"
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Array [
        Object {
          "backgroundColor": "#ccc",
          "borderRadius": 6,
          "height": 150,
          "width": 90,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 16,
          "marginLeft": 5,
          "marginTop": 10,
        },
      ]
    }
  />
</View>
`;
